protocol/synthetix/contracts/storage/AccountDelegationIntents.sol

```diff
@@ -0,0 +1,149 @@
+        accountDelegationIntents = loadValid(id);
+        if (accountDelegationIntents.accountId == 0) {
+            // Uninitialized storage will have a 0 accountId
+            accountDelegationIntents.accountId = id;
+        }
+    }
+
+    function addIntent(Data storage self, DelegationIntent.Data storage delegationIntent) internal {
+        self.intentsId.add(delegationIntent.id);
+        self
+            .intentsByPair[
+                keccak256(abi.encodePacked(delegationIntent.poolId, delegationIntent.accountId))
+            ]
+            .add(delegationIntent.id);
+
+        self.netDelegatedAmountPerCollateral[delegationIntent.collateralType] += delegationIntent
+            .deltaCollateralAmountD18;
+
+        if (!self.delegatedCollaterals.contains(delegationIntent.collateralType)) {
+            self.delegatedCollaterals.add(delegationIntent.collateralType);
+        }
+    }
+
+    function removeIntent(
+        Data storage self,
+        DelegationIntent.Data storage delegationIntent
+    ) internal {
+        if (!self.intentsId.contains(delegationIntent.id)) {
+            return;
+        }
+
+        self.intentsId.remove(delegationIntent.id);
+        self
+            .intentsByPair[
+                keccak256(abi.encodePacked(delegationIntent.poolId, delegationIntent.accountId))
+            ]
+            .remove(delegationIntent.id);
+
+        self.netDelegatedAmountPerCollateral[delegationIntent.collateralType] -= delegationIntent
+            .deltaCollateralAmountD18;
+    }
+
+    function getIntent(
+        Data storage self,
+        uint256 intentId
+    ) internal view returns (DelegationIntent.Data storage) {
+        if (!self.intentsId.contains(intentId)) {
+            revert DelegationIntent.InvalidDelegationIntentId();
+        }
+        return DelegationIntent.load(intentId);
+    }
+
+    /**
+     * @dev Returns the delegation intent stored at the specified nonce id.
+     */
+    function intentIdsByPair(
+        Data storage self,
+        uint128 poolId,
+        uint128 accountId
+    ) internal view returns (uint256[] memory intentIds) {
+        return self.intentsByPair[keccak256(abi.encodePacked(poolId, accountId))].values();
+    }
+
+    /**
+     * @dev Cleans all intents related to the account. This should be called upon liquidation.
+     */
+    function cleanAllExpiredIntents(Data storage self) internal {
+        uint256[] memory intentIds = self.intentsId.values();
+        for (uint256 i = 0; i < intentIds.length; i++) {
+            DelegationIntent.Data storage intent = DelegationIntent.load(intentIds[i]);
+            if (intent.intentExpired()) {
+                removeIntent(self, intent);
+            }
+        }
+    }
+
+    /**
+     * @dev Cleans all intents related to the account. This should be called upon liquidation.
+     */
+    function cleanAllIntents(Data storage self) internal {
```

Comment:
About this function, i remember speaking to jared about it, and remember him saying, that it would be possible to nuke it all, by just deleting the key, rather than iterating.

Reply:
added nukability